### PR TITLE
test(client): add `TestIsUnavailableErr`

### PR DESCRIPTION
**Pull request by wafuwafu13**
Please read https://github.com/etcd-io/etcd/blob/main/CONTRIBUTING.md#contribution-flow.

### CODE REVIEW

File: client/v3/client_test.go [hunk 1]
```diff
@@ -152,23 +155,63 @@ func TestDialNoTimeout(t *testing.T) {
...
+		true,
+		"cancel on context should be halte error",
+	)
+}
+
+func TestIsUnavailableErr(t *testing.T) {
+	assert.Equal(t,
+		isUnavailableErr(context.TODO(), errors.New("etcdserver: some etcdserver error")),
+		false,
+		`error prefixed with "etcdserver: " should not be unavailable error by default`,
```
[the remark this answers is not in the document]
Comment (ahrtr) [reply client/v3/client_test.go:191]: There are some examples errors which prefixed with `etcdserver` and are `codes.Unavailable`.

```
	ErrGRPCStopped                    = status.New(codes.Unavailable, "etcdserver: server stopped").Err()
	ErrGRPCTimeout                    = status.New(codes.Unavailable, "etcdserver: request timed out").Err()
	ErrGRPCTimeoutDueToLeaderFail     = status.New(codes.Unavailable, "etcdserver: request timed out, possibly due to previous leader failure").Err()
	ErrGRPCTimeoutDueToConnectionLost = status.New(codes.Unavailable, "etcdserver: request timed out, possibly due to connection lost").Err()
	ErrGRPCTimeoutWaitAppliedIndex    = status.New(codes.Unavailable, "etcdserver: request timed out, waiting for the applied index took too long").Err()
	ErrGRPCUnhealthy                  = status.New(codes.Unavailable, "etcdserver: unhealthy cluster").Err()
```

File: client/v3/client_test.go [hunk 2]
```diff
@@ -152,23 +155,63 @@ func TestDialNoTimeout(t *testing.T) {
...
+		"no error and active context should be halt error",
+	)
 	cancel()
-	if !isHaltErr(ctx, nil) {
-		t.Errorf("cancel on context should be Halted")
-	}
+	assert.Equal(t,
+		isHaltErr(ctx, nil),
+		true,
+		"cancel on context should be halte error",
```
Comment (ptabor) [inline client/v3/client_test.go:183]: nit: halte->halt

File: client/v3/client_test.go [hunk 3]
```diff
@@ -152,23 +155,63 @@ func TestDialNoTimeout(t *testing.T) {
...
-	if !isHaltErr(context.TODO(), errors.New("etcdserver: some etcdserver error")) {
-		t.Errorf(`error prefixed with "etcdserver: " should be Halted by default`)
-	}
-	if isHaltErr(context.TODO(), rpctypes.ErrGRPCStopped) {
-		t.Errorf("error %v should not halt", rpctypes.ErrGRPCStopped)
-	}
-	if isHaltErr(context.TODO(), rpctypes.ErrGRPCNoLeader) {
-		t.Errorf("error %v should not halt", rpctypes.ErrGRPCNoLeader)
-	}
+	assert.Equal(t,
```
Comment (ptabor) [inline client/v3/client_test.go:158]: I would slightly prefer to make it an 'table driven' test, as the pattern is very repetitive. 
But the current form is an already improvement. Thank you.

https://dave.cheney.net/2013/06/09/writing-table-driven-tests-in-go

**Review by ptabor — APPROVED** [review]

**Review by ahrtr — APPROVED** [review]
LGTM

File: client/v3/client_test.go (hunk 3 above)
Comment (ahrtr) [reply client/v3/client_test.go:158]: Please feel free to resolve this comment in a separate PR.

**Pull request merged**